site-builder/src/lib/client/queries.ts

```diff
@@ -0,0 +1,125 @@
+import { expect } from "utils/assert";
+
+import { SITE_ENV } from "../env/site";
+import { getClient } from "./client";
+
+export const getPubType = async (name: string | string[]) => {
+	const nameArray = Array.isArray(name) ? name : [name];
+
+	const pubTypeId = await getClient().pubTypes.getMany({
+		params: {
+			communitySlug: SITE_ENV.COMMUNITY_SLUG,
+		},
+		query: { name: nameArray },
+	});
+
+	if (pubTypeId.status !== 200) {
+		console.error(pubTypeId.body);
+		throw new Error("Failed to fetch pub type");
+	}
+
+	return expect(pubTypeId.body?.filter((pubType) => nameArray.includes(pubType.name)));
+};
+
+export const getJournal = async (opts?: {
+	depth?: number;
+	withRelatedPubs?: boolean;
+	fieldSlugs?: string[];
+}) => {
+	const journalPubType = await getPubType("Journal");
+
+	const journal = await getClient().pubs.getMany({
+		params: {
+			communitySlug: SITE_ENV.COMMUNITY_SLUG,
+		},
+		query: {
+			pubTypeId: [expect(journalPubType[0].id)],
+			limit: 1,
+			depth: opts?.depth ?? 3,
+			withRelatedPubs: opts?.withRelatedPubs ?? true,
+		},
+	});
+
+	if (journal.status !== 200) {
+		throw new Error("Failed to fetch journal");
+	}
+
+	return expect(journal.body?.[0]);
+};
```

Comment:
so obviously we shouldnt do it like this, but pass in the id of the journal or something in the action (it is the pub that the action is run on).
none of this should be treated as how we should _actually_ do it